### PR TITLE
Big updoot of the new year: monolithic unatomic PR edition

### PR DESCRIPTION
**Pull request by Michiyamenotehifunana**
## About The Pull Request
This PR will:
1) Port Hardsuit ricochets
2) Port various combat hitchance and cover (Low wall and tables) improvements
3) Port combat diving and leaping (Rest while moving in a direction, C is the new hotkey for resting)
4) Port new sound effects for melee stuff
5) Rework shields for the fourth time
6) Rework wall penetration so that each wall penetrated weakens the offending projectile
7) Add craftable throwing knives
8) Turn the mekhane shield into a mini bag that holds only swords

Refer to the Eris PR numbers listed in the commits for more details.

## Why It's Good For The Game
More content! More fun!

## Changelog
```changelog Toriate
add: Ported a ton of combat stuff from upstream, including combat dives!
```

### CODE REVIEW

**Comment by Shadow-Quill** [comment]
Please atomize this.

**Comment by Michiyamenotehifunana** [comment]
I refuse.

File: code/game/machinery/deployable.dm [hunk 1]
```diff
@@ -132,13 +132,48 @@ for reference:
...
+		return 1
+	if (get_dist(P.starting, loc) <= 1) //Cover won't help you if people are THIS close
+		return 1
+	if (get_turf(P.original) == cover)
+		var/valid = FALSE
+		var/distance = get_dist(P.last_interact,loc)
+		P.check_hit_zone(loc, distance)
+
+		var/targetzone = check_zone(P.def_zone)
+		if (targetzone in list(BP_R_LEG, BP_L_LEG, BP_GROIN))
```
Comment (Mocha-Not-Latte) [inline code/game/machinery/deployable.dm:159]: This doesnt cover the feet

Comment (Jamini) [reply code/game/machinery/deployable.dm:159]: Eris doesn't have feet and hands. That's why.

**Comment by Jamini** [comment]
Right, executive decision time. The RIG deflection chance needs to go. Wholesale. We can re-implement it later as a module. Right now it's simply ridiculous, especially when coupled with a shield.

File: code/game/machinery/deployable.dm (hunk 1 above)
Comment (RadiantFlash7) [reply code/game/machinery/deployable.dm:159]: Wha. I thought Eris was old bay code. Bay code had hands/feet..

**Review by Jamini — CHANGES_REQUESTED** [review]
RIG deflect needs to go.
Deployable shields need to cover the feet
We should probably get the fallthrough PR as part of this, so we don't have people stuck in space.

**Comment by fludd12** [comment]
👀 merge conflicts
but other than that this is looking good, thank you for addressing the ricochet issue.

**Comment by Jamini** [comment]
Merge conflict is with his other icon/sprites PR. As long as we favor that file it should be fine.

**Comment by Jamini** [comment]
Issues with combat knives that need to be resolved:

1. Knives have no miss chance
2. Knives can be fired faster than bullets
3. No cooldown on thrown
4. Every knife embeds

This is not going to work for us.

**Comment by Jamini** [comment]
Issues with diving

1. Diving into objects does no damage
2. You can break windows and doors via diving headbuts
3. Diving is instant

**Comment by RadiantFlash7** [comment]
> Issues with diving
> 
>3. Diving is instant

*Instant as in, theres no cooldown.

**Comment by Mocha-Not-Latte** [comment]
> Issues with diving
> 
> 1. Diving into objects does no damage
> 2. You can break windows and doors via diving headbuts
> 3. Diving is instant
> 4. Combat diving allows you to change direction in space.

I had the zoomies :)

**Comment by Jamini** [comment]
This PR needs significant work to be considered for approval and due to this and other events is being closed for the time being.

**Pull request closed without merging**